source/read/retrieve.txt

```diff
@@ -0,0 +1,227 @@
+   section of this guide.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the :ref:`<pymongo-quickstart>`.
+
+.. _pymongo-retrieve-find:
+
+Find Documents
+--------------
+
+{+driver-short+} includes two methods for retrieving documents from a collection:
+``find_one()`` and ``find()``.
+These methods take a **query filter** and return one or more matching documents.
+A query filter is an object that specifies the documents you want to retrieve in
+your query. 
+
+.. TODO: To learn more about query filters, see :ref:`pymongo-specify-query`.
+
+.. _pymongo-retrieve-find-one:
+
+Find One Document
+~~~~~~~~~~~~~~~~~
+
+To find a single document in a collection, call the ``find_one()`` method and pass a query
+filter that specifies the criteria of the document you want to find.
+If more than one document matches the query
+filter, this method returns the *first* matching document from the retrieved
+results as a Python dictionary. If no documents match the query filter, the method returns
+``None``.
+It is useful when you know there is only one matching document, or are only interested in the first match.  
+
+The following examples use the ``find_one()`` method to find the first document where
+the ``"cuisine"`` field has the value ``"Bakery"``. Select the corresponding tab to see
+an example in the Python shell or a ``.py`` file. 
+
+.. tabs::
+
+   .. tab:: Shell
+      :tabid: find-one-shell
+
+      .. code-block:: python
+         :copyable: true
+
+         >>> sample_restaurants.restaurants.find_one({"cuisine": "Bakery"})
+
+   .. tab:: File
+      :tabid: find-one-file
+
+      .. code-block:: python
+         :copyable: true
+
+         restaurant = sample_restaurants.restaurants.find_one({"cuisine": "Bakery"})
+
+.. tip:: Sort Order
+
+   The ``find_one()`` method returns the first document in 
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+.. To learn more about sorting, see the TODO: sorting page.
+
+.. TODO: To see a full example of using the ``find_one()`` method to find a single document, see
+   :ref:`pymongo-retrieve-additional-information`.
+
+.. _pymongo-retrieve-find-multiple:
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``find()``
+method that specifies the criteria of the documents you want to retrieve.
+
+You can use a **cursor** to iterate over the documents returned by the ``find()``
+method. A cursor is a mechanism that allows an application to iterate over database 
+results while holding only a subset of them in memory at a given time. Cursors
+are useful when your ``find()`` method returns a large amount of documents.
```

Comment:
S: I think this paragraph should be moved under the first code example (on line 134 where it says: "You can iterate over the documents..."). That way it separates using the find method from iterating over the results of the find method, and also moves the first mention of a cursor closer to the first example of using one.

Reply:
great idea